### PR TITLE
Fixed  #1677 TheThe color of Navigation bar can't be set to primary color even if the toggle button is off.

### LINKED ISSUE #1677: Bug in theme
**Actual Behaviour**

The color of Navigation bar can still be set to the primary color even if the toggle button is off.

**Expected Behaviour**

The color of the navigation bar should be set to the primary color if the toggle button is off.

**Steps to reproduce it**

Go to the settings activity and turn off the toggle button for the "Colored Navigation Bar" . Now try to select a different primary color, the color of the navigation bar can be set to the primary color.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by raj10071997**
Fixed #1677

Changes: Previously the color of the Bottom Navigation bar changes according to the primary color even if the toggle button of "Colored Navigation Bar" remained off. So made some changes to ensure that the color of the navigation bar is not set to present primary color if the toggle button is off. If the toggle button is off, then the color of the navigation bar is same as the primary color just before turning off the toggle button and remains the same till it is turned on and set to the present primary color. 

Screenshots for the change: 
navigationbar

### CODE REVIEW

File: app/src/main/java/org/fossasia/phimpme/base/ThemedActivity.java [hunk 1]
```diff
@@ -61,8 +61,11 @@ public void updateTheme() {
         coloredNavBar = SP.getBoolean(getString(R.string.preference_colored_nav_bar), true);
         obscuredStatusBar = SP.getBoolean(getString(R.string.preference_translucent_status_bar), true);
         applyThemeImgAct = SP.getBoolean(getString(R.string.preference_apply_theme_pager), true);
-        setNavigationBarColor(getPrimaryColor());
-        setNavBarColor();
+        if(coloredNavBar)
+        {
```
Comment (abishekvashok) [inline app/src/main/java/org/fossasia/phimpme/base/ThemedActivity.java]: I've it inline

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -586,6 +586,7 @@
     <string name="folder">Folder</string>
     <string name="color_palette">Color Palette</string>
     <string name="app_intro_skip">Introduction Skipped!</string>
+    <string name="preference_BottomNavColor">getBottomNavBarColor</string>
```
Comment (abishekvashok) [inline app/src/main/res/values/strings.xml]: Pls follow the similar trend that is using power case letters in strong names.

File: app/src/main/java/org/fossasia/phimpme/base/ThemedActivity.java (hunk 1 above)
Comment (raj10071997) [reply app/src/main/java/org/fossasia/phimpme/base/ThemedActivity.java]: how can you  inline it without else statement?

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (raj10071997) [reply app/src/main/res/values/strings.xml]: I don't understand you here. Please explain.

File: app/src/main/java/org/fossasia/phimpme/base/ThemedActivity.java (hunk 1 above)
Comment (raj10071997) [reply app/src/main/java/org/fossasia/phimpme/base/ThemedActivity.java]: Ok, the curly bracket and the if statement should be on the same line, I thought you were talking about the concept of inline functions and that got me confused.

**Comment by raj10071997** [comment]
@abishekvashok made the changes as suggested by you, please review it again.

**Comment by abishekvashok** [comment]
Can you check the failing tests pls?

**Comment by raj10071997** [comment]
Actually right now I am away from my laptop , will be seeing to this issue
soon.

On Jan 26, 2018 11:41 AM, "Abishek V Ashok" <notifications@github.com>
wrote:

Can you check the failing tests pls?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/fossasia/phimpme-android/pull/1727#issuecomment-360693467>,
or mute the thread

**Comment by raj10071997** [comment]
@abishekvashok  all the tests have been passed. Can you please review it again ? 
@anantprsd5 @mohitmanuja Please review this PR.

**Comment by isuruAb** [comment]
Please resolve conflicts

**Comment by raj10071997** [comment]
@isuruAb Conflicts have been resolved. Please review it again.

**Pull request merged**